app/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -111,8 +111,9 @@ static void kscan_gpio_irq_callback_handler(const struct device *dev, struct gpi
 
 static void kscan_gpio_timer_handler(struct k_timer *timer) {
     struct kscan_gpio_data *data = CONTAINER_OF(timer, struct kscan_gpio_data, poll_timer);
+    const struct kscan_gpio_config *cfg = data->dev->config;
 
-    kscan_gpio_direct_queue_read(&data->work, 0);
+    kscan_gpio_direct_queue_read(&data->work, cfg->debounce_period);
```

Comment:
I'm not sure this is 100% right. For polling mode we have a fixed timer if every 10ms, which them fires this callback. Until we.have the more nuanced debounce work from @joelspadin perhaps that timer interval should honor debounce period instead?

Reply:
Ah okie. I will close this now if there is advanced debounce is on the work